src/lib.rs

```diff
@@ -412,6 +424,162 @@ fn run_for_context<TContext: Sync>(
     })
 }
 
+pub fn run_for_slice_with_callback<'a>(
+    slice: impl Into<RopeOrSlice<'a>>,
```

Comment:
So I think the idea here will be that the `slice` argument can be generic as something like `impl TextProvider<'a> + Parseable` (so not need to have any "specific" awareness of `ropey` at the `tree_sitter_grep` level)

But that is blocked by running into lifetime issues at the point of calling into `tree_sitter` (per https://github.com/tree-sitter/tree-sitter/pull/2432)

(although as discussed maybe we could reasonably temporarily depend on a forked version of `tree_sitter` if there is in fact a way to tell all of the individual language-grammar crates "hey use this version of that dependency instead"?)

So for now made it take the "concrete" type of `RopeOrSlice`

(the point of this is to try and be more efficient where in `tree-sitter-lint-lsp` and maybe will also add to the `tree-sitter-lint` "fixing loop" we're storing file text in a `ropey::Rope` and so it's presumably expensive to convert that to "one big" `&[u8]` vs letting it be parsed/queried "in chunks")